### PR TITLE
ci: upgrade knip@5.32.0

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:

~Refs https://github.com/eslint/eslint/issues/18990, temporarily pins knip to 5.30.6 until https://github.com/eslint/eslint/issues/18990 is resolved.~

Fixes https://github.com/eslint/eslint/issues/18990: updates knip dependency to skip v5.31.0.

#### What changes did you make? (Give an overview)

Updated package.json.

### CODE REVIEW

File: package.json
Comment (amareshsm) [inline package.json:170]:
```suggestion
    "knip": "^5.32.0",
```
knip version [`5.32.0`](https://github.com/webpro-nl/knip/releases/tag/5.32.0) has been released, and I don't see any issues that are referenced on the ESLint issue thread: - https://github.com/eslint/eslint/issues/18990 Can we proceed with migrating directly to knip `^5.32.0`?

**Review by amareshsm — APPROVED** [review]
LGTM. Thanks!

Waiting for 2nd approval.

**Review by fasttime — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
